test/e2e-api/kafka.go

```diff
@@ -35,49 +35,116 @@ type KafkaController struct {
 	adminClient sarama.ClusterAdmin
 	producer    sarama.SyncProducer
 	consumer    sarama.Consumer
+	mLock       sync.RWMutex
 }
 
-func NewKafkaController() *KafkaController {
-	// initialize Kafka handlers
-	var brokers = []string{bootstrapServers}
+/*
+  getKafkaClient is used for getting a Kafka admin client for the given controller config. It is implemented with a lazy loading mechanism:
+  1) A new client is created only for the first request
+  2) Returning the current client if it exists
+*/
+
+func (n *KafkaController) getKafkaClient() sarama.ClusterAdmin {
```

Comment:
Since this is just for testing, we don't need to separate them into multiple different `get` functions, instead, do something like `func (n *KafkaController) getProducerAndConsumer() (sarama.ClusterAdmin, sarama.SyncProducer, sarama.Consumer)`.

This is to avoid too much redundant code.